src/query/src/query_engine/options.rs

```diff
@@ -0,0 +1,139 @@
+// Copyright 2023 Greptime Team
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use common_catalog::consts::DEFAULT_CATALOG_NAME;
+use datafusion_common::TableReference;
+use session::context::QueryContextRef;
+use snafu::ensure;
+
+use crate::error::{QueryAccessDeniedSnafu, Result};
+
+#[derive(Default)]
+pub struct QueryOptions {
+    pub validate_table_references: bool,
+}
+
+pub fn validate_catalog_and_schema(
+    catalog: Option<&str>,
+    schema: &str,
+    query_ctx: &QueryContextRef,
+) -> Result<()> {
+    ensure!(
+        (catalog.is_none() || catalog.unwrap() == query_ctx.current_catalog())
```

Comment:
If catalog is none, do we need to check if current_catalog is the default catalog? If not, the catalog in error message below will be wrong.

Reply:
if catalog is none, should we put "" to err msg, e.g: "Illegal access to catalog: and schema: xxx"
or change `QueryAccessDenied` to receive a err msg and format it every time?

Reply:
We need to check how we deal with the situation when catalog is none:

- if it uses `DEFAULT_CATALOG_NAME` instead, check if query_ctx.current_catalog() == DEFAULT_CATALOG_NAME
- if it uses catalog from query_context, simply change error message to use query_ctx.current_catalog

Reply:
only `ShowTables` would call `validate_param` with `None` as catalog, and it actually use `query_ctx.current_catalog()` later on.

so I think the signature can be change to `fn validate_param(catalog: Option<&str>, schema: &str, query_ctx: &QueryContextRef)` and use `query_ctx.current_catalog()` directly